### PR TITLE
feat: store identity id with identity's public key as a DB key

### PR DESCRIPTION
**Pull request by jawid-h**

### CODE REVIEW

File: lib/dpp/DriveStateRepository.js [hunk 1]
```diff
@@ -46,6 +49,15 @@ class DriveStateRepository {
     const transaction = this.getDBTransaction('identity');
 
     await this.identityRepository.store(identity, transaction);
+
+    const storeMapPromises = identity.getPublicKeys()
```
Comment (shumkov) [inline lib/dpp/DriveStateRepository.js:53]: We need to store the only first key (master key). Also, probably we need separate methods to store/fetch identity? We need the fetch method for the first public key uniqueness validation.

Comment (jawid-h) [reply lib/dpp/DriveStateRepository.js:53]: @shumkov 

> Also, probably we need separate methods to store/fetch identity? 

Could you, please, explain a bit more where do you want them, why and what those methods do.

>  We need the fetch method for the first public key uniqueness validation.

Could you, please, explain a bit more.

File: lib/dpp/DriveStateRepository.js [hunk 2]
```diff
@@ -46,6 +49,12 @@ class DriveStateRepository {
     const transaction = this.getDBTransaction('identity');
 
     await this.identityRepository.store(identity, transaction);
+
+    const [firstPublicKey] = identity.getPublicKeys();
+
+    await this.publicKeyIdentityIdMapLevelDBRepository.store(
```
Comment (shumkov) [inline lib/dpp/DriveStateRepository.js:55]: We need to store it in a separate method `storePublicKeyIdentityId`, also we need `fetchPublicKeyIdentityId` for Identity Create Transition validation function.

File: lib/dpp/DriveStateRepository.js (hunk 1 above)
Comment (shumkov) [reply lib/dpp/DriveStateRepository.js]: We need to store it in a separate method storePublicKeyIdentityId that will be called from the apply state transition function, also we need fetchPublicKeyIdentityId for Identity Create Transition validation function.

Comment (jawid-h) [reply lib/dpp/DriveStateRepository.js]: @shumkov  done

**Pull request merged**